### DISCUSSION TITLE
No DOCTYPE?

### DISCUSSION THREAD

**Discussion by VAggrippino** [cssbattle/forum#69] [Bugs reports]
I created a solution that renders differently on CodePen than it does on CSSBattle, or even in static file opened directly by the browser.
Check this out: https://codepen.io/VAggrippino/pen/GRWPMbj

In Chrome and Firefox it looked just like CSSBattle, so I was ready to blame CodePen, but I'm glad I took a minute to figure it out... If I remove `<!DOCTYPE html>` from the source of CodePen's result, it looks like it does on CSSBattle... The tests are done in _Quirks Mode_! 😲

I don't imagine this occurs very often, but I think it might lead to a bit of frustration if someone tries to use something they learned while perfecting their CSSBattle solution.

**Comment by SelenIT** [comment]
I consider the fact that CSSBattle playground doesn't add anything to the submitted code, including doctype, a feature rather than a bug. It's  mentioned [in the FAQ](https://cssbattle.dev/faqs/#how-to-play). Some Quirks-Mode-specific "optimizations" (like omitting the `px` unit in values of CSS1 properties) are listed among [the Tips&Tricks](https://cssbattle.dev/tips/) (and widely used in the top scoring solutions).

Since nearly all code golfing competitions are kind of "hacking" games, I believe that the educational value of CSSBattle is to teach "what _can happen_ in HTML/CSS" rather than "what you _can use_ in CSS". It provides a safe environment to play with things that unlikely would occur in the everyday work following the good practices (like DOM tree not following the markup structure because of implied optional HTML tags), but just knowing that such things exist (and what can cause them) can be extremely useful for debugging. Knowing some secrets of the "dark side" of HTML/CSS doesn't mean you should (ab)use them, it also can be helpful to avoid and prevent them from happening in your code.

**Reply by VAggrippino** [reply]
Okay, I get it. I thought it was a mistake, but since it's intentional and documented I guess there's really not much more to say about it.
